server/routerlicious/packages/services-core/src/mongoDocumentRepository.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FindAndOperateDocumentMode, ICollection, IDb, IDocumentRepository } from "./database";
+import { IDocument } from "./document";
+
+export class MongoDocumentRepository implements IDocumentRepository {
+    constructor(private readonly collection: ICollection<IDocument>) {
+    }
+
+    public static create(
+        database: IDb,
+        collectionName: string,
+    ): MongoDocumentRepository {
+        return new MongoDocumentRepository(database.collection<IDocument>(collectionName));
+    }
+
+    async readDocument(filter: any): Promise<IDocument> {
+        return this.collection.findOne(filter);
+    }
+
+    async updateDocument(
+        filter: any,
+        updates: any,
+        options?: any,
+    ): Promise<void> {
+        await (options.upsert
+            ? this.collection.upsert(filter, updates, options?.addToSet)
```

Comment:
>options?.addToSet [](http://example.com/codeflow?start=54&length=17)

Why not pass options as is?

Reply:
It means I have to modify the api and would cause breaking changes:
<img width="609" alt="image" src="https://user-images.githubusercontent.com/3719246/222162474-1fc9282b-b4ce-409b-8211-35c1ec31de63.png">

Initially I thought `addToSet` was a boolean so I didn't do it. But now checked again found this parameter is any so it should be OK for me to rename `addToSet` to `options`, thoughts?